packages/webapp/src/util/task.ts

```diff
@@ -172,6 +172,24 @@ export const formatSoilAmendmentProductToDBStructure = (
   });
 };
 
+export const formatSoilAmendmentTaskToDBStructure = (
+  soilAmendmentTask: FormSoilAmendmentTask,
+  { methods }: { methods: SoilAmendmentMethod[] },
+): DBSoilAmendmentTask => {
```

Comment:
I'm sorry for the confusion, but `FormSoilAmendmentTask` and `DBSoilAmendmentTask` are for the structure one level above:
```
{..., soil_amendment_task: {}, soil_amendment_task_products: {})
```

I don't really know the appropriate names for them, but the new types need to be defined for `soil_amendment_task`.

Reply:
My typescript is not great -- so I tried to copy what you had for task products almost exatcly. The naming to me implies I should just expand the existing types and they should all still be accurate? I will try..

Reply:
Okay @SayakaOno I tried something. My only thought was if I should add conditional to soil_amendment_task_products in DBTask and FormTask;

Reply:
A conditional check to ensure `soil_amendment_task_products` exists?


Your changes look good, thank you!
But I still see some TS warnings, so I will open a PR to address them. If you are not seeing them, it's hard to address...

![Screenshot 2024-07-18 at 10 21 29 AM](https://github.com/user-attachments/assets/3cb1d651-2d3a-4b9c-a905-01dc4140f6f0)

Reply:
Conditional check on DBTask type to make it not required (since it is loosely typing Task object which may not be soil amendment)

Oh yeah those went away for me! I did not want to add structured clone but it solved those typings from me. Thanks for finding! What typescript version is your intellisense?

Reply:
Ah I got it! DBTask is only used for soil amendment task, so I think it's okay for now. But I would like to revisit it and handle it properly...

How can I check my intellisense? 🙈

Reply:
Bottom right corner of VSCode - we probably want to match webapp package version.

![Screenshot 2024-07-18 at 2 38 44 PM](https://github.com/user-attachments/assets/09405b06-815d-4014-8c9b-45bcf76f8b89)

Reply:
🙏 Mine is 5.4.5!